### PR TITLE
feat: Add Session type in SessionDetailActivity (#2342)

### LINKED ISSUE #2342: Add session Type in session details activity
**Actual Behaviour**

Sesssion details activity doesn't mention anything about the type of the session.

**Expected Behaviour**

The type of the session should be displayed

**Steps to reproduce it**

Select any session from the sessions of any speaker

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AkshayCHD**
Fixes #2342 

Changes: The type of session is now visible in SessionDetailActivity
screenshot_1518905232

### CODE REVIEW

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

File: android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java [hunk 1]
```diff
@@ -112,6 +113,13 @@
     protected ImageView youtubeThumbnail;
     @BindView(R.id.watch)
     protected ImageButton playButton;
+    @BindView(R.id.sessionType)
+    protected TextView session_type;
+    @BindView(R.id.dot)
+    protected TextView dot;
+    @BindView(R.id.type)
+    protected TextView type_textview;
+
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java]: That's not how you should do it. Define only one textview. If you need to hide it, wrap the layout in LinearLayout

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java (hunk 1 above)
Comment (AkshayCHD) [reply android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java]: @iamareebjamal but if i do that the grid view will distort, the gridview has three columns in every row,
If i create two dummy textviews after the LinearLayout then the positioning of ':' will not be proper.
Either i should create a separate issue moving from grid layout to linear layout, as in the current code already there is a problem that when the track name is empty only ':' would be displayed.

Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java]: You can use framelayout as a wrapper. Check if that is possible

File: android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java [hunk 2]
```diff
@@ -112,6 +113,10 @@
     protected ImageView youtubeThumbnail;
     @BindView(R.id.watch)
     protected ImageButton playButton;
+    @BindView(R.id.sessionType)
+    protected TextView session_type;
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java]: Java follows camel case style

File: android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java [hunk 3]
```diff
@@ -207,6 +213,15 @@ private void updateSession() {
             text_room1.setText(location);
         }
 
+        SessionType type = session.getSessionType();
+
+        if(type != null) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java]: Syntax issue

File: android/app/src/main/res/layout/content_session_detail.xml [hunk 4]
```diff
@@ -67,6 +67,42 @@
...
+                    <LinearLayout
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_columnSpan="3"
+                        android:id="@+id/typeLayout"
+                        android:layout_rowSpan="1">
+
+                        <TextView
+                            android:id="@+id/type"
+                            android:layout_width="62sp"
```
Comment (iamareebjamal) [inline android/app/src/main/res/layout/content_session_detail.xml]: Never hardcode width, and sp is used for text size

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

File: android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java [hunk 5]
```diff
@@ -207,6 +215,15 @@ private void updateSession() {
             text_room1.setText(location);
         }
 
+        SessionType type = session.getSessionType();
+
+        if (type != null) {
+            location = type.getName();
+            sessionType.setText(type.getName());
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java]: Show the type layout if it is present, hide by default

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by AkshayCHD** [comment]
@iamareebjamal please review, there has been no change in the UI so the screenshots have not been updated

File: android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java [hunk 6]
```diff
@@ -207,6 +215,14 @@ private void updateSession() {
             text_room1.setText(location);
         }
 
+        SessionType type = session.getSessionType();
+
+        if (type != null) {
+            location = type.getName();
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionDetailActivity.java]: Why is location set to type name?

**Comment by AkshayCHD** [comment]
@iamareebjamal removed that, sorry

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**